osu.Game.Rulesets.Mania/Objects/Drawables/DrawableHoldNote.cs

```diff
@@ -254,6 +254,8 @@ protected override void Update()
                 float yOffset = Direction.Value == ScrollingDirection.Up ? -Y : Y;
                 sizingContainer.Height = 1 - yOffset / DrawHeight;
             }
+            else
+                sizingContainer.Height = 1;
```

Comment:
I understand the intention, but applied like this, this breaks behaviour when a hold note is released.

Current master:

https://github.com/ppy/osu/assets/20418176/ae6e7f99-99d7-4526-87ae-da41a00aac7b

This PR:

https://github.com/ppy/osu/assets/20418176/33b075a2-230d-4051-a727-199d9db0fd1f

Note how the note snaps back to full height on release.

I'm guessing that the intent here was closer to:

```suggestion
            else if (Time.Current < HitObject.StartTime)
                sizingContainer.Height = 1;
```

Reply:
Yeah, huge oversight.

I've applied something similar to your diff with a bit more restructuring to avoid the same conditional repeating twice.